tlsutil/config_test.go

```diff
@@ -613,166 +1106,7 @@ func TestConfigurator_LoadCAs(t *testing.T) {
 	}
 }
 
-func TestConfigurator_CommonTLSConfigInsecureSkipVerify(t *testing.T) {
-	c, err := NewConfigurator(Config{}, nil)
-	require.NoError(t, err)
-	tlsConf := c.commonTLSConfig(false)
-	require.True(t, tlsConf.InsecureSkipVerify)
-
-	require.NoError(t, c.Update(Config{VerifyServerHostname: false}))
-	tlsConf = c.commonTLSConfig(false)
-	require.True(t, tlsConf.InsecureSkipVerify)
-
-	require.NoError(t, c.Update(Config{VerifyServerHostname: true}))
-	tlsConf = c.commonTLSConfig(false)
-	require.False(t, tlsConf.InsecureSkipVerify)
-}
-
-func TestConfigurator_CommonTLSConfigPreferServerCipherSuites(t *testing.T) {
-	c, err := NewConfigurator(Config{}, nil)
-	require.NoError(t, err)
-	tlsConf := c.commonTLSConfig(false)
-	require.False(t, tlsConf.PreferServerCipherSuites)
-
-	require.NoError(t, c.Update(Config{PreferServerCipherSuites: false}))
-	tlsConf = c.commonTLSConfig(false)
-	require.False(t, tlsConf.PreferServerCipherSuites)
-
-	require.NoError(t, c.Update(Config{PreferServerCipherSuites: true}))
-	tlsConf = c.commonTLSConfig(false)
-	require.True(t, tlsConf.PreferServerCipherSuites)
-}
-
-func TestConfigurator_CommonTLSConfigCipherSuites(t *testing.T) {
-	c, err := NewConfigurator(Config{}, nil)
-	require.NoError(t, err)
-	tlsConf := c.commonTLSConfig(false)
-	require.Empty(t, tlsConf.CipherSuites)
-
-	conf := Config{CipherSuites: []uint16{
-		tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305}}
-	require.NoError(t, c.Update(conf))
-	tlsConf = c.commonTLSConfig(false)
-	require.Equal(t, conf.CipherSuites, tlsConf.CipherSuites)
-}
```

Comment:
I think this is covered by the `CipherSuites` case in `TestConfigurator_IncomingConfig_Common`, which exercises it with a full handshake rather than checking the `tls.Config`.

https://github.com/hashicorp/consul/blob/dda057259133a625330c8f9eaafac8f8f78f5dca/tlsutil/config_test.go#L70-L91

You're talking about the CHACHA20_POLY1305 suites, right? I'm not sure if that value in the original test is relevant at all (or more likely randomly selected). I don't know enough about crypto stuff to say for sure, but I think you're right we should support them if Go does.

Reply:
> I'm not sure if that value in the original test is relevant at all

It was somewhat relevant in that specifying that specific suite would (I believe) not pass through Consul's config [`ParseCiphers`](https://github.com/hashicorp/consul/blob/ee0e8bf50034de2b2ce13b933a1d6bf14c54403c/tlsutil/config.go#L947-L980) allow-list to be used in the actual TLS connection established - the test appeared to be testing a configuration that wouldn't actually work.

Would it be possible to add a test here for the negative case of specifying some older valid but insecure/unsupported cipher suite and ensure it _isn't_ used for the actual TLS connection? (I'm not sure if the correct behavior is for the connection to fail to be established, for it to error early, or for `CipherSuites` to fall back to defaults as if it was unspecified.)

Reply:
Oooooh I see, thanks for calling that out! Will add a test case to capture that 🙇🏻‍♂️

Reply:
Actually, having given this a bit more thought, I think we'd be testing that `CipherSuites` rejects the unsupported value. Which may be better served by a unit test on `CipherSuites` rather than a full-stack test like these — sorry I was a little slow on the uptake there! 😅